src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -1793,55 +1910,34 @@ def lint_line_length(
                 desired_indent = current_indent
                 if target_balance >= 1:
                     desired_indent += single_indent
+                target_breaks = matched_indents[target_balance]
                 reflow_logger.debug(
                     "    Targeting balance of %s, indent: %r for %s",
                     target_balance,
                     desired_indent,
-                    matched_indents[target_balance],
+                    target_breaks,
                 )
-                line_results: List[LintResult] = []
-                for e_idx in matched_indents[target_balance]:
-                    # If the option is the final element. Don't touch it, because
-                    # there's already an indent there.
-                    if e_idx == i:
-                        continue
 
-                    e = cast(ReflowPoint, elements[e_idx])
-
-                    # We need to check for negative sections so they get the right
-                    # indent (otherwise they'll be over indented).
-                    # The `desired_indent` above is for the "uphill" side.
-                    following_class_types = elements[e_idx + 1].class_types
-                    indent_stats = e.get_indent_impulse(
-                        allow_implicit_indents, following_class_types
+                # Is one of the locations the final element? If so remove it.
+                # There's already a line break there.
+                if i in target_breaks:
+                    target_breaks.remove(i)
+
+                # Is it an "integer" indent or a fractional indent.
+                # Handle accordingly.
+                if target_balance % 1 == 0:
```

Comment:
This seems wrong. **Any** integer returns 0 when you do `% 1` on it.

Reply:
ah yes - but fractional values don't. I'll update the comments to be clearer.

Reply:
I've updated the comment to:
```sql
                # Is it an "integer" indent or a fractional indent?
                # Integer indents (i.e. 1.0, 2.0, ...) are based on Indent and
                # Dedent tokens. Fractional indents (i.e. 1.5, 1.52, ...) are
                # based more on rebreak spans (e.g. around commas and operators).
                # The latter is simpler in that it doesn't change the indents,
                # just adds line breaks. The former is more complicated.
                # NOTE: Both of these methods mutate the `elem_buffer`.
                if target_balance % 1 == 0:
                    ...
```

Reply:
Sounds good. There's also `math.modf()`, but this way is fine now that you added the comment. I never think about using `%` with floats. 🤯